scheduler/plugin/plugin_test.go

```diff
@@ -1350,6 +1350,176 @@ func TestPostBind(t *testing.T) {
+		pod                   *v1.Pod
+		nodeName              string
+		expectedStatusCode    framework.Code
+		expectedStatusMessage string
+	}{
+		{
+			name:               "Success",
+			args:               []string{"test", "addPod"},
+			pod:                test.PodSmall,
+			nodeName:           "good",
+			expectedStatusCode: framework.Success,
+		},
+		{
+			name:                  "Error",
+			args:                  []string{"test", "addPod"},
+			pod:                   test.PodSmall,
+			nodeName:              "bad",
+			expectedStatusCode:    framework.Error,
+			expectedStatusMessage: "name is bad",
+		},
+		{
+			name:               "min statusCode",
+			guestURL:           test.URLTestPreFilterExtensionsFromGlobal,
+			pod:                test.PodSmall,
+			nodeName:           test.NodeSmall.Name,
+			globals:            map[string]int32{"status_code": math.MinInt32},
+			expectedStatusCode: math.MinInt32,
+		},
+		{
+			name:               "max statusCode",
+			guestURL:           test.URLTestPreFilterExtensionsFromGlobal,
+			pod:                test.PodSmall,
+			nodeName:           test.NodeSmall.Name,
+			globals:            map[string]int32{"status_code": math.MaxInt32},
+			expectedStatusCode: math.MaxInt32,
+		},
+		{
+			name:               "panic",
+			guestURL:           test.URLErrorPanicOnPreFilterExtensions,
+			pod:                test.PodSmall,
+			nodeName:           test.NodeSmall.Name,
+			expectedStatusCode: framework.Error,
+			expectedStatusMessage: `wasm: prefilterextensions error: panic!
+wasm error: unreachable
+wasm stack trace:
+	panic_on_prefilterextensions.$1() i32`,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			guestURL := tc.guestURL
+			if guestURL == "" {
+				guestURL = test.URLTestBind
+			}
+
+			p, err := wasm.NewFromConfig(ctx, "wasm", wasm.PluginConfig{GuestURL: guestURL, Args: tc.args})
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer p.(io.Closer).Close()
+
+			pl := wasm.NewTestWasmPlugin(p)
+			if len(tc.globals) > 0 {
+				pl.SetGlobals(tc.globals)
+			}
+			pl.CreateGuestInBindingGuestPool(tc.pod.UID)
+
+			status := p.(framework.PreFilterExtensions).AddPod(ctx, nil, tc.pod, nil, nil)
+			if want, have := tc.expectedStatusCode, status.Code(); want != have {
+				t.Fatalf("unexpected status code: want %v, have %v", want, have)
+			}
+			if want, have := tc.expectedStatusMessage, status.Message(); want != have {
+				t.Fatalf("unexpected status message: want %v, have %v", want, have)
+			}
+		})
+	}
+}
+
+func TestRemovePod(t *testing.T) {
+	tests := []struct {
+		name                  string
+		guestURL              string
+		args                  []string
+		globals               map[string]int32
+		pod                   *v1.Pod
+		nodeName              string
+		expectedStatusCode    framework.Code
+		expectedStatusMessage string
+	}{
+		{
+			name:               "Success",
+			args:               []string{"test", "removePod"},
```

Comment:
It's a bit complicated, but this arg means the arg in guest/testdata/filter/main.go.
So I guess you need to use `preFilterExtensions` instead of `removePod`.

```
func main() {
	// Multiple tests are here to reduce re-compilation time and size checked
	// into git.
	var plugin extensionPoints = noopPlugin{}
	if len(os.Args) == 2 {
		switch os.Args[1] {
		case "filter":
			plugin = filterPlugin{}
		case "preFilter":
			plugin = preFilterPlugin{}
		case "postFilter":
			plugin = postFilterPlugin{}
		case "preFilterExtensions":
			plugin = preFilterExtensions{}
		}
	}
	prefilter.SetPlugin(plugin)
	filter.SetPlugin(plugin)
	postfilter.SetPlugin(plugin)
	prefilterextensions.SetPlugin(plugin)
}
```

Reply:
I understood.